File: tests/consumer/src/extras/java/eu/darken/porter/sdk/consumer/ExtrasBridge.java
package eu.darken.porter.sdk.consumer;

import android.os.IBinder;

import eu.darken.porter.sdk.extras.PorterSystemProperties;
import eu.darken.porter.sdk.extras.PorterSystemServices;

/** Compiles only in the extras build: keeps both of that artifact's classes off the dead-code path. */
public class ExtrasBridge {
    public int sdkInt() {
        return PorterSystemProperties.getInt("ro.build.version.sdk", 0);
    }

    public IBinder packageManager() {
        return PorterSystemServices.getSystemService("package");
    }
}

File: tests/consumer/src/extras/java/eu/darken/porter/sdk/consumer/ExtrasBridge.kt
package eu.darken.porter.sdk.consumer

import android.os.IBinder
import eu.darken.porter.sdk.Porter
import eu.darken.porter.sdk.extras.PorterSystemServices
import eu.darken.porter.sdk.extras.getSystemPropertyInt

/** Compiles only in the extras build: keeps both of that artifact's entry points off the dead-code path. */
class ExtrasBridge {

    fun sdkInt(): Int? = Porter.connection.value?.getSystemPropertyInt("ro.build.version.sdk", 0)

    fun packageManager(): IBinder? = PorterSystemServices.getSystemService("package")
}
